File: fixed/CPP-PATH-001/fixed_path_traversal.cpp
// CPP-PATH-001 fixed-version fixture for patch-diff evaluation.
#include <stdexcept>
#include <string>

static std::string suspiciousJoin(const std::string& requested) {
    return "./allowed/" + requested;
}

std::string safeJoin(const std::string& requested) {
    if (requested.find("..") != std::string::npos || requested.find('/') != std::string::npos) {
        throw std::runtime_error("invalid path");
    }
    return suspiciousJoin(requested);
}

File: fixed/RUST-PATH-TRAVERSAL-001/fixed_path_traversal.rs
// RUST-PATH-TRAVERSAL-001 fixed-version fixture for patch-diff evaluation.
use std::path::{Component, Path, PathBuf};

pub fn resolve_report_path(requested: &str) -> Result<PathBuf, &'static str> {
    let path = Path::new(requested);
    if path.components().any(|part| matches!(part, Component::ParentDir | Component::RootDir | Component::Prefix(_))) {
        return Err("invalid path");
    }
    Ok(PathBuf::from("./reports").join(path))
}
